components/ip_selector_sync_test.go

```diff
@@ -0,0 +1,56 @@
+package components
+
+import (
+	"os"
+	"testing"
+
+	sp "github.com/scipipe/scipipe"
+)
+
+func TestIPSelectorSync(t *testing.T) {
+	wf := sp.NewWorkflow("testwf", 4)
+
+	// Create a non-empty file
+	p1 := wf.NewProc("p1", "echo foo > {o:foo}")
+	p1.SetOut("foo", ".tmp/foo.txt")
+
+	// Create an EMPTY file
+	p2 := wf.NewProc("p2", "touch {o:bar}")
+	p2.SetOut("bar", ".tmp/bar.txt")
+
+	// This function should return true for any files to be INCLUDED and false
+	// for any files that should be DROPPED
+	filterEmptyFilesFunc := func(ip *sp.FileIP) bool {
+		return ip.Size() != 0
```

Comment:
Will this really work in case the file has size 0? Doesn't this only look for an empty IP? But isn't the IP the file name and that will never be empty?

Reply:
The `Size()` method does a check of the physical file, see https://github.com/scipipe/scipipe/blob/master/ip.go#L154-L159

Also, the test works ... and I did try breaking the test by creating a non-empty file, and the test failed as expected :)